### PR TITLE
[DEV-7420] Add default Data table Header to Charts

### PR DESCRIPTION
**Pull request by Atash3000**

### CODE REVIEW

File: packages/core/components/EditorPanel/DataTableEditor.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ const DataTable: React.FC<DataTableProps> = ({ config, updateField, isDashboard,
           }
         />
       )}
-      <TextField value={config.table.indexLabel} section='table' fieldName='indexLabel' label='Index Column Header' updateField={updateField} />
+      <TextField value={config.table.indexLabel ? config.table.indexLabel : config.xAxis?.dataKey} section='table' fieldName='indexLabel' label='Index Column Header ddddd' updateField={updateField} />
```
Comment (adamdoe) [inline packages/core/components/EditorPanel/DataTableEditor.tsx:78]: @Atash3000 please remove the ddddd part here. I'm also wondering if we should use the data key at all. Most data keys that I've seen aren't the most readable and I'm not sure we want to output that to the user. If I remember correctly, the ticket shared that its okay to output `<span aria-hidden>No title</span>` as a backwards compatible option.

**Review by adamdoe — CHANGES_REQUESTED** [review]

File: packages/core/components/EditorPanel/DataTableEditor.tsx (hunk 1 above)
Comment (Atash3000) [reply packages/core/components/EditorPanel/DataTableEditor.tsx:78]: If data key is not readable user can override it using Index header input under Data Table section.

File: packages/core/components/EditorPanel/DataTableEditor.tsx [hunk 2]
```diff
@@ -75,7 +75,7 @@ const DataTable: React.FC<DataTableProps> = ({ config, updateField, isDashboard,
           }
         />
       )}
-      <TextField value={config.table.indexLabel} section='table' fieldName='indexLabel' label='Index Column Header' updateField={updateField} />
+      <TextField value={config.table.indexLabel ? config.table.indexLabel : config.xAxis?.dataKey} section='table' fieldName='indexLabel' label='Index Column Header' updateField={updateField} />
```
Comment (adamdoe) [inline packages/core/components/EditorPanel/DataTableEditor.tsx:78]: I think what we need here is the empty string still. Then, in the table header output we can check for the empty string and if it is empty we can output the content that's mentioned in the ticket `<th><span sr-only>Not Applicable</span></th>`

**Review by adamdoe — CHANGES_REQUESTED** [review]

**Comment by adamdoe** [comment]
@Atash3000 this PR has feedback on it.

**Pull request closed without merging**